test/components/connect.spec.js

```diff
@@ -949,6 +949,38 @@ describe('React', () => {
       expect(decorated.refs.wrappedInstance.someInstanceMethod()).toBe(someData)
     })
 
+    it('should do no parsing when the response body is empty', (done) => {
+      window.fetch = (request) => {
+        return new Promise((resolve) => {
+          resolve(new window.Response('', { status: 204 }))
+        })
+      }
+
+      @connect(() => ({ testFetch: `/empty` }))
+      class Container extends Component {
+        render() {
+          return <Passthrough {...this.props} />
+        }
+      }
+
+      const container = TestUtils.renderIntoDocument(
+        <Container />
+      )
+
+      const init = TestUtils.findRenderedComponentWithType(container, Container)
+      expect(init.state.data.testFetch).toIncludeKeyValues(
+        { fulfilled: false, pending: true, reason: null, refreshing: false, rejected: false, settled: false, value: null }
+      )
+
+      setImmediate(() => {
+        const fulfilled = TestUtils.findRenderedComponentWithType(container, Container)
+        expect(fulfilled.state.data.testFetch).toIncludeKeyValues(
+          { fulfilled: true, pending: false, reason: null, refreshing: false, rejected: false, settled: true, value: null }
+        )
+        done()
+      })
+    })
+
```

Comment:
> ... but I see the full object

Yup, that's my point. My [previous post](https://github.com/heroku/react-refetch/pull/68#discussion_r53899088) was an abbreviation to help clarify my [initial comment](https://github.com/heroku/react-refetch/pull/68#discussion_r53890223), but I guess I missed the mark. :smile: 

All I was trying to say is that I think the failure should be clearer and less verbose, and take you right to the heart of the problem. It's a minor thing, though.